src/core/amount.js

```diff
@@ -148,20 +156,22 @@ Amount.prototype.subtract = function(subtrahend) {
 
 // XXX Diverges from cpp.
 Amount.prototype.multiply = function(multiplicand) {
-  var multiplicandAmount = Amount.from_json(multiplicand);
-  // TODO: probably should just multiply by multiplicandAmount._value
-  var multiplyBy = multiplicandAmount.is_native() ?
-    multiplicandAmount._value.times(Amount.bi_xns_unit)
-    : multiplicandAmount._value;
-  return this._copy(this._value.times(multiplyBy));
+
+  const multiplicandAmount = Amount.from_json(multiplicand);
+  const multiplyBy = multiplicandAmount.is_native() ?
+    new Value_XRP(multiplicandAmount._value, Amount.bi_xns_unit)
+    : new Value_IOU(multiplicandAmount._value);
+  const thisValue = new Value_IOU(this._value);
+  return this._copy(new BigNumber(thisValue.multiply(multiplyBy)._value));
```

Comment:
If you go down this path I agree with what Chris says above. If the amount class `has no mention of` of any BigNumber, it follows that the `._value` member should be one of these new *Value classes that encapsulate them.


Anyway, it just occurred to me, that there's no amount class for the new ripple api, that uses counterparty instead of issuer.